cab-token-generator/java/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactory.java

```diff
@@ -151,6 +394,48 @@ public Builder setSourceCredential(GoogleCredentials sourceCredential) {
       return this;
     }
 
+    /**
+     * Sets the minimum acceptable lifetime for a generated CAB token.
+     *
+     * <p>This value determines the minimum remaining lifetime required on the intermediate token
+     * before a CAB token can be generated. If the intermediate token's remaining lifetime is less
+     * than this value, CAB token generation will be blocked and a refresh will be initiated. This
+     * ensures that generated CAB tokens have a sufficient lifetime for use.
+     *
+     * @param minimumTokenLifetime The minimum acceptable lifetime for a generated CAB token. Must
+     *     be positive.
+     * @return This {@code Builder} object.
+     * @throws IllegalArgumentException if minimumTokenLifetime is negative or zero.
+     */
+    @CanIgnoreReturnValue
+    public Builder setMinimumTokenLifetime(Duration minimumTokenLifetime) {
+      if (minimumTokenLifetime.isNegative() || minimumTokenLifetime.isZero()) {
+        throw new IllegalArgumentException("Minimum token lifetime must be positive.");
```

Comment:
nit: I think `must be greater than zero` sounds better (Ignore if this is official verbiage)